closed/adds/jdk/src/share/native/jdk/crypto/jniprovider/NativeCrypto.c

```diff
@@ -318,16 +353,107 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
         (NULL == OSSL_RSA_private_encrypt) ||
         (NULL == OSSL_BN_bin2bn) ||
         (NULL == OSSL_BN_set_negative) ||
-        (NULL == OSSL_BN_free)) {
-        //fprintf(stderr, "One or more of the required symbols are missing in the crypto library\n");
-        //fflush(stderr);
+        (NULL == OSSL_BN_free) ||
+        ((NULL == OSSL_CRYPTO_num_locks) && (0 == ossl_ver)) ||
+        ((NULL == OSSL_CRYPTO_THREADID_set_numeric) && (0 == ossl_ver)) ||
+        ((NULL == OSSL_OPENSSL_malloc) && (0 == ossl_ver)) ||
+        ((NULL == OSSL_CRYPTO_THREADID_set_callback) && (0 == ossl_ver)) ||
+        ((NULL == OSSL_CRYPTO_set_locking_callback) && (0 == ossl_ver))) {
+        // fprintf(stderr, "One or more of the required symbols are missing in the crypto library\n");
+        // fflush(stderr);
         unload_crypto_library(handle);
         return -1;
     } else {
+        if (0 == ossl_ver) {
+            if (0 != thread_setup()) {
+                unload_crypto_library(handle);
+                return -1;
+            }
+        }
         return 0;
     }
  }
 
+#ifdef WINDOWS
+static HANDLE *lock_cs;
+
+int thread_setup()
+{
+    int i;
+    int size = (*OSSL_CRYPTO_num_locks)() * sizeof(HANDLE);
```

Comment:
The value of `(*OSSL_CRYPTO_num_locks)()` should be in a local variable so it can be reused in the `for` loop below.
The type of `size` should be `size_t`.